themes/src/main/resources/theme/base/admin/resources/js/controllers/users.js

```diff
@@ -527,6 +527,27 @@ module.controller('UserDetailCtrl', function($scope, realm, user, BruteForceUser
         }
     }
 
+    function bfs(tree, key, collection) {
+    	if (!tree[key] || tree[key].length === 0) return;
+    	for (var i=0; i < tree[key].length; i++) {
+    		var child = tree[key][i]
+    		collection.push(child);
+    		bfs(child, key, collection);
+    	}
+    	return;
+    }
+
+    function flattenGroups(groups) {
+        var flattenedGroups = [];
+        if (!groups || groups.length === 0) return groups;
+        for (var i=0; i < groups.length; i++) {
+            flattenedGroups.push(groups[i]);
+            bfs(groups[i], "subGroups", flattenedGroups);
```

Comment:
I don't think you need to pass in "subGroups".  I don't think that value can ever change.

Reply:
Fixed in my latest commit.